controllers/lookup/k8s_client.go

```diff
@@ -0,0 +1,160 @@
+package lookup
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/datainfrahq/druid-operator/apis/druid/v1alpha1"
+	internalhttp "github.com/datainfrahq/druid-operator/pkg/http"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"net/http"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type K8sClient struct {
+	client client.Client
+}
+
+func NewK8sClient(client client.Client) K8sClient {
+	return K8sClient{
+		client: client,
+	}
+}
+
+func (c *K8sClient) FindLookups(ctx context.Context, reports map[types.NamespacedName]Report) (map[ClusterKey]map[LookupKey]Spec, error) {
+	specs := &v1alpha1.DruidLookupList{}
+	if err := c.client.List(ctx, specs); err != nil {
+		return nil, err
```

Comment:
You should take into account when getting "IsNotFound" error type. If this object been deleted bafore the controller got to reconcile it, returnin error might lead to infinite reconciles of non-existing object.

Reply:
Here I'm attempting to get a complete view of all the Lookups present in the cluster, which is why I'm using `List` instead of `Get`. Could `List` also return a `IsNotFound` error?

The reasoning for behind this can both be found in the PR description and as response to https://github.com/datainfrahq/druid-operator/pull/150#discussion_r1620454673.

Reply:
You are right :)